### ISSUE TITLE
ICE running `cargo doc` on `typenum` on beta

### ISSUE THREAD

**Issue by ErichDonGubler** [rust-lang/rust#54655]
I'm getting an ICE trying to run `cargo doc` on https://github.com/paholg/typenum/commit/284f456a707229bdbcac5a951d6e8b138a5abcc3:

```
~/…/pull-requests/typenum master$ cargo doc
   Compiling typenum v1.10.0 (C:\Users\egubler\workspace\personal\pull-requests\typenum)
 Documenting typenum v1.10.0 (C:\Users\egubler\workspace\personal\pull-requests\typenum)
error: internal compiler error: librustc\traits\structural_impls.rs:178: impossible case reached

thread '<unnamed>' panicked at 'Box<Any>', librustc_errors\lib.rs:586:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.30.0-beta.8 (269c1bdc7 2018-09-27) running on x86_64-pc-windows-msvc

error: Could not document `typenum`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name typenum 'src\lib.rs' --color always -o 'C:\Users\egubler\workspace\personal\pull-requests\typenum\target\doc' -L 'dependency=C:\Users\egubler\workspace\personal\pull-requests\typenum\target\debug\deps'` (exit code: 1)
```

Not sure why this is happening, but it's consistent, so I figured I should report this here.

**Comment by Havvy** [comment]
This is a duplicate of #53505
